### DISCUSSION TITLE
vips_affine: should default oarea include specified offsets?

### DISCUSSION THREAD

**Discussion by lovell** [libvips/libvips#3454] [General]
The [vips_affine docs](https://www.libvips.org/API/current/libvips-resample.html#vips-affine) say:

> By default `oarea` is just large enough to cover the whole of the transformed input image.

As originally reported at https://github.com/lovell/sharp/issues/3632, using the following image as input:

![](https://user-images.githubusercontent.com/114157573/232445943-402891f0-e673-476c-9dc9-b97fd30b65d1.png)

```
vips affine in.png out.png "1 0 0.5 1"
```
...produces the expected:

![](https://user-images.githubusercontent.com/210965/233415203-3f19849d-3678-4092-bbcf-54a6d6a52533.png)

...but adding an offset without an `oarea` does not extend the default output area.
```
vips affine in.png out.png "1 0 0.5 1" --odx 32.3 --ody 44.5
```
![](https://user-images.githubusercontent.com/210965/233415486-c329c4d3-e9db-422a-8aba-4ea5f07e1ecb.png)

Should libvips include offsets in the default `oarea` calculation, or should we update the docs to state that the use of offsets may also necessitate a custom `oarea`?

**Comment by seranus** [comment]
I have a similar issue but the opposite, I want to keep the image size and crop everything outside without having to do 2 pipelines.

Similar to the issue [https://github.com/lovell/sharp/issues/2488](url)

I'd appreciate of we could manually set `oarea` or have an option to keep the original viewport

In my case doing additional resizing on offsets would further increase the intermediate image memory size
